messageBus/amazonSQS.js

```diff
@@ -0,0 +1,55 @@
+const AWS = require('aws-sdk');
+const config = require('./aws.config');
+
+AWS.config.update({
+  accessKeyId: process.env.SQS_ACCESS_KEY || config.accessKey,
+  secretAccessKey: process.env.SQS_SECRET_KEY || config.secretKey,
+});
+
+const sqs = new AWS.SQS({ region: process.env.SQS_REGION || config.region });
+
+const MAX_NUMBER_OF_MESSAGES_TO_RECEIVE = 10;
+const MESSAGE_VISIBILITY_TIMEOUT = 10;
+const subscribers = [
+  config.queue,
+];
+
+module.exports.publish = (message) => {
+  subscribers.forEach((subscriber) => {
+    const sqsParams = {
```

Comment:
These should really be wrapped in a promise, and waited to make sure they are all fulfilled. Right now things just keep going on even if there are errors thrown.

Reply:
The http server that serves the client app search requests publishes these messages. Given that the priority for that server is to handle the requests, publishing these messages adopts a "fire-and-forget" approach (i.e., the publisher/httpServer will not wait because these messages are for analytics purposes because they are not business critical). However, I think in the next iteration of this code, I will wrap it in a promise to catch the errors and log them to a file so that in the future, a process can consume this file to catch messages that for some reason failed to be sent to the message bus. Thanks!